libethashseal/Ethash.cpp

```diff
@@ -158,9 +158,18 @@ void Ethash::verifyTransaction(ImportRequirements::value _ir, TransactionBase co
 		else
 			_t.checkChainId(-4);
 	}
-	// Unneeded as it's checked again in Executive. Keep it here since tests assume it's checked.
-	if (_ir & ImportRequirements::TransactionBasic && _t.baseGasRequired(evmSchedule(EnvInfo(_bi))) > _t.gas())
-		BOOST_THROW_EXCEPTION(OutOfGasIntrinsic());
+	if (_ir & ImportRequirements::TransactionBasic)
```

Comment:
That function is called by the `verifyBlock` , 
but seems transactions that comes from `onPeerTransaction` get accepted and thus relayed even if the `chainId` is wrong.
Would it make sense to check for the replay protection also at the 'network' layer?